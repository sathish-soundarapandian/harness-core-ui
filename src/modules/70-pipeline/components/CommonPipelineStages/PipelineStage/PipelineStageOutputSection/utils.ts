/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { AllowedTypes, getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { FormikErrors, FormikProps } from 'formik'
import { get, isEmpty, set } from 'lodash-es'
import type { UseStringsReturn } from 'framework/strings'
import type { PipelineStageOutputs } from 'services/pipeline-ng'

export const MAX_LENGTH = 64

export interface PipelineStageOutputSectionProps {
  children: React.ReactElement
}

export interface PipelineStageOutputData {
  outputs: PipelineStageOutputs[]
}

export interface OutputPanelInputSetProps {
  template: PipelineStageOutputData
  path: string
  allowableTypes: AllowedTypes
  readonly?: boolean
}

export interface ConectedOutputPanelInputSetProps extends OutputPanelInputSetProps {
  formik: FormikProps<PipelineStageOutputData>
}

interface ValidateOutputPanelInputSetProp {
  data: PipelineStageOutputData
  template?: PipelineStageOutputData
  getString?: UseStringsReturn['getString']
}

export const validateOutputPanelInputSet = ({
  data,
  template,
  getString
}: ValidateOutputPanelInputSetProp): FormikErrors<PipelineStageOutputData> => {
  const errors: FormikErrors<PipelineStageOutputData> = { outputs: [] }
  data?.outputs?.forEach((output: PipelineStageOutputs, index: number) => {
    const currentVariableTemplate = get(template, `outputs[${index}].value`, '')
    if (isEmpty(output.value) && getMultiTypeFromValue(currentVariableTemplate) === MultiTypeInputType.RUNTIME) {
      set(errors, `outputs[${index}].value`, getString?.('fieldRequired', { field: output.name }))
    }
  })
  return errors
}
