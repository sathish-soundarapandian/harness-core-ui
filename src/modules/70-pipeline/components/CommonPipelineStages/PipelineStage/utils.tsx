/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

/* eslint-disable react/function-component-definition */
import React from 'react'
import type { CellProps, Renderer } from 'react-table'
import {
  Text,
  FontVariation,
  Layout,
  Icon,
  Color,
  Popover,
  ButtonSize,
  ButtonVariation,
  Button,
  IconName
} from '@harness/uicore'
import { defaultTo, get } from 'lodash-es'
import { Classes, PopoverInteractionKind, Position } from '@blueprintjs/core'
import { useParams } from 'react-router-dom'
import { String, useStrings } from 'framework/strings'
import type { PMSPipelineSummaryResponse } from 'services/pipeline-ng'
import { useAppStore } from 'framework/AppStore/AppStoreContext'
import { StoreType } from '@common/constants/GitSyncTypes'
import routes from '@common/RouteDefinitions'
import type { PipelineType, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import css from './PipelineStageMinimalMode.module.scss'

interface CodeSourceWrapper {
  textName: string
  iconName: IconName
  size: number
}

export const PipelineNameIdTagCell: Renderer<CellProps<PMSPipelineSummaryResponse>> = ({ row }) => {
  const data = row.original
  return (
    <Layout.Horizontal spacing="xsmall">
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H6 }} lineClamp={1} color={Color.BLACK}>
          {data?.name}
        </Text>
        <Text font={{ variation: FontVariation.BODY }} lineClamp={1} color={Color.GREY_600}>
          <String stringID="idLabel" vars={{ id: data?.identifier }} />
        </Text>
      </Layout.Vertical>
      {Object.keys(get(data, 'tags', {})).length > 0 && (
        <>
          <Icon name="main-tags" size={15} />
          <Text color={Color.GREY_600}>{Object.keys(get(data, 'tags', {})).length}</Text>
        </>
      )}
    </Layout.Horizontal>
  )
}

export const CodeSourceCell: Renderer<CellProps<PMSPipelineSummaryResponse>> = ({ row }) => {
  const { gitDetails } = row.original
  const { getString } = useStrings()
  const data = row.original
  const { isGitSyncEnabled: isGitSyncEnabledForProject, gitSyncEnabledOnlyForFF } = useAppStore()
  const isGitSyncEnabled = isGitSyncEnabledForProject && !gitSyncEnabledOnlyForFF
  const isRemote = data.storeType === StoreType.REMOTE || isGitSyncEnabled
  const inlineWrapper: CodeSourceWrapper = {
    textName: getString('inline'),
    iconName: 'repository',
    size: 10
  }
  const remoteWrapper: CodeSourceWrapper = {
    textName: getString('repository'),
    iconName: 'remote-setup',
    size: 12
  }

  return (
    <div className={css.codeSourceColumnContainer}>
      <Popover
        disabled={!isRemote}
        position={Position.TOP}
        interactionKind={PopoverInteractionKind.HOVER}
        className={Classes.DARK}
        content={
          <Layout.Vertical spacing="small" padding="large" className={css.contentWrapper}>
            <Layout.Horizontal spacing="small" flex={{ alignItems: 'center', justifyContent: 'start' }}>
              <Icon name="github" size={14} color={Color.GREY_200} />
              <Text color={Color.WHITE} font={{ variation: FontVariation.SMALL }}>
                {get(gitDetails, 'repoName', get(gitDetails, 'repoIdentifier'))}
              </Text>
            </Layout.Horizontal>
            {gitDetails?.filePath && (
              <Layout.Horizontal spacing="small" flex={{ alignItems: 'center', justifyContent: 'start' }}>
                <Icon name="remotefile" size={14} color={Color.GREY_200} />
                <Text color={Color.WHITE} font={{ variation: FontVariation.SMALL }}>
                  {gitDetails.filePath}
                </Text>
              </Layout.Horizontal>
            )}
          </Layout.Vertical>
        }
      >
        <div className={css.codeSourceColumn}>
          <Icon
            name={isRemote ? remoteWrapper.iconName : inlineWrapper.iconName}
            size={isRemote ? remoteWrapper.size : inlineWrapper.size}
            color={Color.GREY_600}
          />
          <Text margin={{ left: 'xsmall' }} font={{ variation: FontVariation.TINY_SEMI }} color={Color.GREY_600}>
            {isRemote ? remoteWrapper.textName : inlineWrapper.textName}
          </Text>
        </div>
      </Popover>
    </div>
  )
}

export const ViewPipelineButtonCell: Renderer<CellProps<PMSPipelineSummaryResponse>> = ({ row, column }) => {
  const { getString } = useStrings()
  const { accountId, module } = useParams<PipelineType<ProjectPathProps>>()
  const data = row.original
  const orgIdentifier = (column as any)?.orgIdentifier
  const projectIdentifier = (column as any)?.projectIdentifier

  const openPipelineInNewTab = (): void => {
    const pipelineStudioPath = routes.toPipelineStudio({
      projectIdentifier,
      orgIdentifier,
      accountId: defaultTo(accountId, ''),
      pipelineIdentifier: get(data, 'identifier', ''),
      module,
      repoIdentifier: get(data, 'gitDetails.repoIdentifier'),
      connectorRef: get(data, 'connectorRef'),
      repoName: get(data, 'gitDetails.repoName'),
      branch: get(data, 'gitDetails.branch'),
      storeType: get(data, 'storeType') as StoreType
    })
    window.open(`${window.location.origin}${window.location.pathname}#${pipelineStudioPath}`, '_blank')
  }

  return (
    <Button
      variation={ButtonVariation.SECONDARY}
      size={ButtonSize.SMALL}
      text={getString('common.viewText')}
      icon="main-view"
      onClick={e => {
        e.preventDefault()
        e.stopPropagation()
        openPipelineInNewTab()
      }}
    />
  )
}
