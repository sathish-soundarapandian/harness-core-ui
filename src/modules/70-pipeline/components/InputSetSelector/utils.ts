/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName, SelectOption } from '@harness/uicore'
import type { EntityGitDetails, InputSetSummaryResponse } from 'services/pipeline-ng'
import css from './InputSetSelector.module.scss'

type InputSetLocal = InputSetSummaryResponse & SelectOption
export interface InputSetValue extends InputSetLocal {
  type: InputSetSummaryResponse['inputSetType']
  gitDetails?: EntityGitDetails
}
export interface ChildPipelineStageProps {
  childOrgIdentifier: string
  childProjectIdentifier: string
  usePortal?: boolean
  inputSetReferences?: string[]
}

export const getIconByType = (type: InputSetSummaryResponse['inputSetType']): IconName => {
  return type === 'OVERLAY_INPUT_SET' ? 'step-group' : 'yaml-builder-input-sets'
}

export const onDragStart = (event: React.DragEvent<HTMLLIElement>, row: InputSetValue): void => {
  event.dataTransfer.setData('data', JSON.stringify(row))
  event.currentTarget.classList.add(css.dragging)
}

export const onDragEnd = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragging)
}

export const onDragLeave = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragOver)
}

export const onDragOver = (event: React.DragEvent<HTMLLIElement>): void => {
  if (event.preventDefault) {
    event.preventDefault()
  }
  event.currentTarget.classList.add(css.dragOver)
  event.dataTransfer.dropEffect = 'move'
}
