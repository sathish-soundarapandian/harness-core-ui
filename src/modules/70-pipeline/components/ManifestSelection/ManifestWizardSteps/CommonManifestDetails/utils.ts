/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ManifestDataType } from '../../Manifesthelper'
import type { ManifestTypes } from '../../ManifestInterface'

export const shouldAllowOnlyOneFilePath = (selectedManifest: ManifestTypes): boolean => {
  return [
    ManifestDataType.ServerlessAwsLambda,
    ManifestDataType.EcsTaskDefinition,
    ManifestDataType.EcsServiceDefinition,
    ManifestDataType.EcsScalableTargetDefinition,
    ManifestDataType.EcsScalingPolicyDefinition,
    ManifestDataType.TasManifest,
    ManifestDataType.TasAutoScaler,
    ManifestDataType.AsgLaunchTemplate,
    ManifestDataType.AsgConfiguration,
    ManifestDataType.GoogleCloudFunctionDefinition,
    ManifestDataType.GoogleCloudFunctionGenOneDefinition,
    ManifestDataType.AwsLambdaFunctionDefinition,
    ManifestDataType.AwsLambdaFunctionAliasDefinition
  ].includes(selectedManifest)
}

/**
 * @description List of manifests which are restricted to single addition
 */
export const allowedManifestForSingleAddition = [ManifestDataType.TasManifest, ManifestDataType.TasAutoScaler]
