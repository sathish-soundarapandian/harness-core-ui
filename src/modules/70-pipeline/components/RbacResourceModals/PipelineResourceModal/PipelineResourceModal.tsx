/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { Container, Layout, Text, Icon } from '@harness/uicore'
import type { CellProps, Renderer } from 'react-table'
import { Color } from '@harness/design-system'
import { Position } from '@blueprintjs/core'
import ReactTimeago from 'react-timeago'
import ResourceHandlerTable, {
  ResourceHandlerTableData
} from '@rbac/components/ResourceHandlerTable/ResourceHandlerTable'
import { PageSpinner, TagsPopover } from '@common/components'
import type { RbacResourceModalProps } from '@rbac/factories/RbacFactory'
import { useStrings } from 'framework/strings'
import { PMSPipelineSummaryResponse, useGetPipelineList } from 'services/pipeline-ng'
import { useMutateAsGet } from '@common/hooks'
import {
  CodeSourceCell,
  LastExecutionCell,
  PipelineNameCell
} from '@pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells'

interface PipelineDTO extends PMSPipelineSummaryResponse {
  admin?: string
  collaborators?: string
  status?: string
}

// eslint-disable-next-line react/function-component-definition
export const RenderColumnPipeline: Renderer<CellProps<PipelineDTO>> = ({ row }) => {
  const rowdata = row.original
  const { getString } = useStrings()

  return (
    <Layout.Vertical spacing="xsmall" data-testid={rowdata.identifier}>
      <Layout.Horizontal spacing="medium">
        <Text
          color={Color.GREY_800}
          tooltipProps={{ position: Position.BOTTOM }}
          tooltip={
            <Layout.Vertical spacing="medium" padding="medium" style={{ maxWidth: 400 }}>
              <Text>{getString('nameLabel', { name: rowdata.name })}</Text>
              <Text>{getString('idLabel', { id: rowdata.identifier })}</Text>
              <Text>{getString('descriptionLabel', { description: rowdata.description })}</Text>
            </Layout.Vertical>
          }
        >
          {rowdata.name}
        </Text>
        {rowdata.tags && Object.keys(rowdata.tags).length ? <TagsPopover tags={rowdata.tags} /> : null}
      </Layout.Horizontal>
      <Text tooltipProps={{ position: Position.BOTTOM }} color={Color.GREY_400}>
        {getString('idLabel', { id: rowdata.identifier })}
      </Text>
    </Layout.Vertical>
  )
}

// eslint-disable-next-line react/function-component-definition
export const RenderLastRunDate: Renderer<CellProps<PipelineDTO>> = ({ row }) => {
  const rowdata = row.original
  const lastRunStartTs = rowdata.recentExecutionsInfo?.[0]?.startTs
  const { getString } = useStrings()
  return (
    <Layout.Vertical spacing="xsmall">
      <Text color={Color.GREY_800}>Last run:</Text>
      <Text color={Color.GREY_400}>
        {lastRunStartTs ? <ReactTimeago date={lastRunStartTs} /> : getString('pipelineSteps.pullNeverLabel')}
      </Text>
    </Layout.Vertical>
  )
}

function PipelineResourceModal({
  searchTerm,
  sortMethod,
  onSelectChange,
  selectedData,
  resourceScope
}: RbacResourceModalProps): React.ReactElement {
  const { accountIdentifier, orgIdentifier = '', projectIdentifier = '' } = resourceScope
  const [page, setPage] = useState(0)
  const { getString } = useStrings()

  const pipelinesQuery = useMutateAsGet(useGetPipelineList, {
    body: {
      filterType: 'PipelineSetup'
    },
    queryParams: {
      accountIdentifier,
      projectIdentifier,
      orgIdentifier,
      searchTerm,
      page,
      size: 10,
      sort: [sortMethod]
    },
    queryParamStringifyOptions: { arrayFormat: 'comma' }
  })

  if (pipelinesQuery.loading) return <PageSpinner />

  const data = pipelinesQuery.data?.data

  return data?.content?.length ? (
    <Container>
      <ResourceHandlerTable
        data={data?.content as ResourceHandlerTableData[]}
        selectedData={selectedData}
        columns={[
          {
            Header: getString('filters.executions.pipelineName'),
            accessor: 'name' as any,
            Cell: PipelineNameCell,
            width: '40%',
            disableSortBy: true
          },
          {
            Header: getString('pipeline.codeSource'),
            accessor: 'storeType' as any,
            Cell: CodeSourceCell,
            width: '20%',
            disableSortBy: true
          },
          {
            Header: getString('pipeline.lastExecution'),
            accessor: 'executionSummaryInfo.lastExecutionTs' as any,
            Cell: LastExecutionCell,
            width: '35%',
            disableSortBy: true
          }
        ]}
        pagination={{
          itemCount: data?.totalElements || 0,
          pageSize: data?.size || 10,
          pageCount: data?.totalPages ?? 1,
          pageIndex: data?.number ?? 0,
          gotoPage: pageNumber => setPage(pageNumber)
        }}
        onSelectChange={onSelectChange}
      />
    </Container>
  ) : (
    <Layout.Vertical flex={{ align: 'center-center' }} spacing="small">
      <Icon name="resources-icon" size={20} />
      <Text font="medium" color={Color.BLACK}>
        {getString('noData')}
      </Text>
    </Layout.Vertical>
  )
}

export default PipelineResourceModal
