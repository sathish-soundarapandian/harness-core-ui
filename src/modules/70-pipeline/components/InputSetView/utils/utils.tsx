/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import {
  AllowedTypes,
  DataTooltipInterface,
  FormInput,
  MultiSelectOption,
  MultiTypeInputType,
  MultiTypeInputValue,
  SelectOption
} from '@harness/uicore'

import type { ServiceSpec } from 'services/cd-ng'
import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'
import { getAllowedValuesFromTemplate } from '@pipeline/utils/CIUtils'

interface UseRenderMultiTypeInputWithAllowedValuesArgs {
  name: string
  tooltipProps?: DataTooltipInterface
  labelKey: string
  placeholderKey?: string
  fieldPath: string
  allowedTypes: AllowedTypes
  template: any
  readonly?: boolean
  className?: string
}

export const useRenderMultiTypeInputWithAllowedValues = ({
  name,
  tooltipProps,
  labelKey,
  placeholderKey,
  fieldPath,
  allowedTypes,
  template,
  readonly,
  className
}: UseRenderMultiTypeInputWithAllowedValuesArgs): { getMultiTypeInputWithAllowedValues: () => JSX.Element } => {
  const { expressions } = useVariablesExpression()

  const getMultiTypeInputWithAllowedValues = (): JSX.Element => {
    const items = getAllowedValuesFromTemplate(template, fieldPath)
    return (
      <FormInput.MultiTypeInput
        name={name}
        label={labelKey}
        useValue
        selectItems={items}
        placeholder={placeholderKey}
        multiTypeInputProps={{
          allowableTypes: allowedTypes,
          expressions,
          selectProps: { disabled: readonly, items }
        }}
        disabled={readonly}
        tooltipProps={tooltipProps}
        className={className}
      />
    )
  }

  return {
    getMultiTypeInputWithAllowedValues
  }
}

interface UseRenderMultiSelectTypeInputWithAllowedValuesArgs {
  name: string
  tooltipProps?: DataTooltipInterface
  labelKey: string
  placeholderKey?: string
  fieldPath: string
  allowedTypes: AllowedTypes
  template: ServiceSpec
  readonly?: boolean
  options: MultiSelectOption[]
  onChange?: (
    value: boolean | string | number | SelectOption | string[] | MultiSelectOption[] | undefined,
    valueType: MultiTypeInputValue,
    type: MultiTypeInputType
  ) => void
}

export const useRenderMultiSelectTypeInputWithAllowedValues = ({
  name,
  tooltipProps,
  labelKey,
  placeholderKey,
  fieldPath,
  allowedTypes,
  template,
  readonly,
  options,
  onChange
}: UseRenderMultiSelectTypeInputWithAllowedValuesArgs): {
  getMultiSelectTypeInputWithAllowedValues: () => JSX.Element
} => {
  const { expressions } = useVariablesExpression()

  const getMultiSelectTypeInputWithAllowedValues = (): JSX.Element => {
    const items = getAllowedValuesFromTemplate(template, fieldPath)
    return (
      <FormInput.MultiSelectTypeInput
        name={name}
        label={labelKey}
        selectItems={items}
        placeholder={placeholderKey}
        disabled={readonly}
        tooltipProps={tooltipProps}
        multiSelectTypeInputProps={{
          multiSelectProps: {
            usePortal: true,
            items: options,
            placeholder: placeholderKey
          },
          allowableTypes: allowedTypes,
          expressions: expressions,
          onChange: onChange
        }}
      />
    )
  }

  return {
    getMultiSelectTypeInputWithAllowedValues
  }
}
