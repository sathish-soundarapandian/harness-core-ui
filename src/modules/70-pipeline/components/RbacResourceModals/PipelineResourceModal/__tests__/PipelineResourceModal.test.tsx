/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, screen } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { defaultAppStoreValues } from '@common/utils/DefaultAppStoreData'
import { pipelines } from '@pipeline/components/PipelineModalListView/__tests__/pipelinelistMocks'
import { SortMethod } from '@common/utils/sortUtils'
import PipelineResourceModal from '../PipelineResourceModal'

const props = {
  searchTerm: '',
  sortMethod: SortMethod.Newest,
  onSelectChange: jest.fn(),
  selectedData: [],
  resourceScope: {
    accountIdentifier: ''
  }
}

const params = {
  accountId: 'accountId',
  orgIdentifier: 'orgIdentifier',
  projectIdentifier: 'projectIdentifier',
  pipelineIdentifier: 'pipelineIdentifier',
  module: 'cd'
}
jest.mock('@common/utils/dateUtils', () => ({
  formatDatetoLocale: (x: number) => x
}))
jest.useFakeTimers()

jest.mock('services/pipeline-ng', () => ({
  useGetPipelineList: jest.fn(() => ({
    mutate: jest.fn().mockResolvedValue(pipelines),
    cancel: jest.fn(),
    loading: false
  }))
}))

describe('PipelineModal List View', () => {
  test('render list view', async () => {
    render(
      <TestWrapper pathParams={params} defaultAppStoreValues={defaultAppStoreValues}>
        <PipelineResourceModal {...props}></PipelineResourceModal>
      </TestWrapper>
    )
    expect(
      await screen.findByRole('link', {
        name: /Sonar Develop/i
      })
    ).toBeInTheDocument()
  })
})
