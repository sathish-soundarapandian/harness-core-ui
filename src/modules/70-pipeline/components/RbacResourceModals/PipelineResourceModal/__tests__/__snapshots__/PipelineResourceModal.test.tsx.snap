// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelineModal List View render list view 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="TableV2--table"
    >
      <div
        class="TableV2--header"
        role="row"
      >
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 5%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            Â 
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 40%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            common.pipeline
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 55%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            lastExecutionTs
          </p>
        </div>
      </div>
      <div
        class="TableV2--body"
      >
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cells"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkBox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 40%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
                data-testid="shell"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                >
                  <span
                    class="bp3-popover-wrapper"
                  >
                    <span
                      class="bp3-popover-target Text--targetClass"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                        tabindex="0"
                      >
                        shell
                      </p>
                    </span>
                  </span>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  idLabel
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 55%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                >
                  Last run:
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  pipelineSteps.pullNeverLabel
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cells"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkBox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 40%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
                data-testid="new_pipeline"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                >
                  <span
                    class="bp3-popover-wrapper"
                  >
                    <span
                      class="bp3-popover-target Text--targetClass"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                        tabindex="0"
                      >
                        new pipeline
                      </p>
                    </span>
                  </span>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  idLabel
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 55%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                >
                  Last run:
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  pipelineSteps.pullNeverLabel
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cells"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkBox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 40%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
                data-testid="pipetagfix"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                >
                  <span
                    class="bp3-popover-wrapper"
                  >
                    <span
                      class="bp3-popover-target Text--targetClass"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                        tabindex="0"
                      >
                        pipe-tag-fix
                      </p>
                    </span>
                  </span>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  idLabel
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 55%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                >
                  Last run:
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  pipelineSteps.pullNeverLabel
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cells"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <label
                class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkBox StyledProps--main"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="bp3-control-indicator"
                />
              </label>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 40%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
                data-testid="httpPipeline"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                >
                  <span
                    class="bp3-popover-wrapper"
                  >
                    <span
                      class="bp3-popover-target Text--targetClass"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                        tabindex="0"
                      >
                        http pipeline
                      </p>
                    </span>
                  </span>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  idLabel
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 55%;"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
                >
                  Last run:
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  pipelineSteps.pullNeverLabel
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
